cmd/zoekt-mirror-gerrit/main.go

```diff
@@ -224,3 +235,41 @@ func marshalBool(b bool) string {
 	}
 	return "0"
 }
+
+func configureGitCredential(rootURL *url.URL, projectURL string, dest string) map[string]string {
+	// Abs path so that we can put it in each git repo config
+	repoDir := filepath.Join(dest, rootURL.Host)
+	gitCredentialStore, err := filepath.Abs(filepath.Join(repoDir, "credentials-store"))
+	if err != nil {
+		log.Fatalf("Absolute dest: %v", err)
+	}
+
+	// credentialValue format: scheme://login:pass@host
+	// XXX: call git credential helper commands
+	cloneURL, err := url.Parse(projectURL)
+	if err != nil {
+		log.Fatalf("url.Parse(): %v", err)
+	}
+	cloneURL.Path = ""
+	cloneURL.User = rootURL.User
+	credentialValue := cloneURL.String()
+
+	if err := os.MkdirAll(repoDir, 0o755); err != nil {
+		log.Fatalf("MkdirAll %s: %v", repoDir, err)
+	}
+	if err := os.WriteFile(gitCredentialStore, []byte(credentialValue), 0644); err != nil {
+		log.Fatalf("WriteFile credentials: %v", err)
+	}
+
+	return map[string]string{
+		"credential.helper": fmt.Sprintf("store --file=%s", gitCredentialStore),
+	}
+}
+
+func anonymousURL(URL *url.URL) string {
+	anonymousURL := *URL
+	if _, has := URL.User.Password(); has {
+		anonymousURL.User = nil
+	}
+	return (&anonymousURL).String()
+}
```

Comment:
what if just user is set? Wouldn't you want to strip it off then? If that is the case this is a simpler implementation:
```suggestion
func anonymousURL(u *url.URL) string {
  anon := *u
  anon.User = nil
  return anon.String()
}
```